adapters/somoaudience/somoaudience.go

```diff
@@ -4,81 +4,158 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/http"
+	"strconv"
 
+	"github.com/golang/glog"
 	"github.com/mxmCherry/openrtb"
 	"github.com/prebid/prebid-server/adapters"
 	"github.com/prebid/prebid-server/errortypes"
 	"github.com/prebid/prebid-server/openrtb_ext"
 )
 
+const config = "hb_pbs_1.0.0"
+
 type SomoaudienceAdapter struct {
 	endpoint string
 }
 
+type somoaudienceReqExt struct {
+	BidderConfig string `json:"prebid"`
+}
+
 func (a *SomoaudienceAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
 
-	totalImps := len(request.Imp)
-	errors := make([]error, 0, totalImps)
-	imp2placement := make(map[string][]int)
+	var errs []error
+	var bannerImps []openrtb.Imp
+	var videoImps []openrtb.Imp
+	var nativeImps []openrtb.Imp
+
+	for _, imp := range request.Imp {
+		// OpenX doesn't allow multi-type imp. Banner takes priority over video.
```

Comment:
Do you share technology with OpenX, or is this a simple cut and paste oversight?